x-pack/plugins/infra/public/components/waffle/index.tsx

```diff
@@ -48,11 +51,27 @@ const calculateBoundsFromMap = (map: InfraWaffleData): InfraWaffleMapBounds => {
 
 export class Waffle extends React.Component<Props, {}> {
   public render() {
-    const bounds = calculateBoundsFromMap(this.props.map);
+    const { loading, map, reload } = this.props;
+    if (loading) {
+      return <InfraLoading height="100%" width="100%" text="Loading data" />;
+    } else if (!loading && map && map.length === 0) {
+      return (
+        <EuiEmptyPrompt
+          title={<h2>There is no data to display.</h2>}
```

Comment:
Not being a native speaker I wonder whether it would be `There are no data [...]` since `data` is plural in itself? :thinking:

Reply:
"There is no data" is grammatically correct in this context. http://www.onlinegrammar.com.au/top-10-grammar-myths-data-is-plural-so-must-take-a-plural-verb/

Reply:
Thanks, TIL about noncount nouns in english! :bulb: